### PR TITLE
Lint: Check for valid parameter IDs in observable/noise parameters

### PR DESCRIPTION
**Pull request by dweindl**
Closes PEtab-dev/libpetab-python#61

### CODE REVIEW

File: petab/measurements.py [hunk 1]
```diff
@@ -200,9 +200,10 @@ def split_parameter_replacement_list(
         delim: delimiter
 
     Returns:
-         List of split values. Numeric values converted to float.
+         List of split values. Numeric values converted to numeric type,
+         parameter IDs as strings.
     """
-    if list_string is None:
+    if list_string is None or list_string == '':
```
Comment (dilpath) [inline petab/measurements.py:206]:
```suggestion
    if list_string is None or list_string.strip() == '':
```

Comment (dilpath) [reply petab/measurements.py:206]: Ah, would need to move after the `isinstance` check below though.

Comment (dweindl) [reply petab/measurements.py:206]: That was actually intentional. I would consider blanks there an error.

File: petab/measurements.py [hunk 2]
```diff
@@ -200,9 +200,10 @@ def split_parameter_replacement_list(
         delim: delimiter
 
     Returns:
-         List of split values. Numeric values converted to float.
+         List of split values. Numeric values converted to numeric type,
+         parameter IDs as strings.
```
Comment (dilpath) [inline petab/measurements.py:204]:
```suggestion
         List of split values. Numeric values may be converted to `float`,
         and parameter IDs are kept as strings.
```

File: petab/measurements.py [hunk 3]
```diff
@@ -212,8 +213,19 @@ def split_parameter_replacement_list(
...
+
+    def convert_and_check(x):
+        x = core.to_float_if_float(x)
+        if isinstance(x, float):
+            return x
+        if lint.is_valid_identifier(x):
+            return x
+        raise ValueError(f"'{x}' in parameter replacement list "
+                         f"'{list_string}' is neither a number, nor a "
+                         "valid parameter ID.")
```
Comment (dilpath) [inline petab/measurements.py:226]: `x` has had `strip()` applied to it, so may be confusing to see possibly just `''`.

```suggestion
        raise ValueError(
            f"The value '{x}' (whitespace removed) in the parameter "
            f"replacement list '{list_string}' is neither a number, nor a "
            "valid parameter ID."
        )
```

**Review by dilpath — APPROVED** [review]
:+1:

File: petab/measurements.py (hunk 3 above)
Comment (dweindl) [reply petab/measurements.py]: Ah, will remove the `strip()`. We don't want that here.

**Review by plakrisenko — APPROVED** [review]

**Pull request merged**